pilot/workflow/generic_hpc.py

```diff
@@ -52,20 +68,150 @@ def run(args):
-
-        # extract user specific info from job report
+        # get the user reference
         user = __import__('pilot.user.%s.common' % args.pilot_user.lower(), globals(), locals(),
                           [args.pilot_user.lower()], -1)
-        # example usage:
-        # user.remove_redundant_files()
-        # user.cleanup_payload()
-        # parse_jobreport_data()
 
-        # implement main function here
-        # ..
+        # get job (and rank)
+        job, rank = resource.get_job(communication_point)
+        # cd to job working directory
+
+        work_dir = resource.set_job_workdir(job, communication_point)
+        work_report['workdir'] = work_dir
+        worker_attributes_file = os.path.join(work_dir, worker_attributes_file)
+        logger.debug("Worker attributes will be publeshied in: {0}".format(worker_attributes_file))
+
+        job.state = 'starting'
+        work_report["jobStatus"] = job.state
+        publish_work_report(work_report, worker_attributes_file)
+
+        # Get HPC specific setup commands
+        logger.info('setup for resource %s: %s' % (args.hpc_resource, str(resource.get_setup())))
+        setup_str = "; ".join(resource.get_setup())
+
+        # Prepare job scratch directory (RAM disk etc.)
+        job_scratch_dir = resource.set_scratch_workdir(job, work_dir)
+
+        my_command = " ".join([job.script, job.script_parameters])
+        my_command = resource.command_fix(my_command, job_scratch_dir)
+        my_command = setup_str + my_command
+
+        # Basic execution. Should be replaced with something like 'run_payload'
+        logger.debug("Going to launch: {0}".format(my_command))
+        logger.debug("Current work directory: {0}".format(job_scratch_dir))
+        payloadstdout = open(payload_stdout_file, "w")
+        payloadstderr = open(payload_stderr_file, "w")
+
+        job.state = 'running'
+        work_report["jobStatus"] = job.state
+        work_report["startTime"] = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+        start_time = time.asctime(time.localtime(time.time()))
+        job.startTime = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+        publish_work_report(work_report, worker_attributes_file)
+
+        stime = time.time()
+        t0 = os.times()
+        exit_code = execute(my_command, stdout=payloadstdout, stderr=payloadstderr, shell=True)
+        t1 = os.times()
+        exetime = time.time() - stime
+        end_time = time.asctime(time.localtime(time.time()))
+        t = map(lambda x, y: x - y, t1, t0)
+        t_tot = reduce(lambda x, y: x + y, t[2:3])
+        job.endTime = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+        payloadstdout.close()
+        payloadstderr.close()
+        if exit_code == 0:
+            job.state = 'finished'
+        else:
+            job.state = 'failed'
+        job.exitcode = exit_code
+
+        work_report["startTime"] = job.startTime
```

Comment:
same comment - you should add these new fields to jobdata class as well, I don't see them there

Reply:
Actually, there is 'joddescription' from minipilot used for the moment. I will see for migration to jobdata.

Reply:
Ok let's leave it for now. Check the parse_job_report instead, as it is too complex so it fails flake8.

Reply:
Yep, i saw these errors from Travis. I hope that 'parse_job_report' will be fixed today, pretty soon. Also i will need review whole 'generic_hpc.py'  because it also complicated.